python/AMI_NOT_PUBLIC_CHECK/AMI_NOT_PUBLIC_CHECK_test.py

```diff
@@ -0,0 +1,216 @@
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+#############
+# Main Code #
+#############
+
+config_client_mock = MagicMock()
+sts_client_mock = MagicMock()
+ec2_client_mock = MagicMock()
+
+class Boto3Mock():
+    def client(self, client_name, *args, **kwargs):
+        if client_name == 'config':
+            return config_client_mock
+        elif client_name == 'sts':
+            return sts_client_mock
+        elif client_name == 'ec2':
+            return ec2_client_mock
+        else:
+            raise Exception("Attempting to create an unknown client")
+
+sys.modules['boto3'] = Boto3Mock()
+
+rule = __import__('AMI_NOT_PUBLIC_CHECK')
+
+# Checks for scenario wherein no non-compliant resources are present
+class CompliantResourcesTest(unittest.TestCase):
+    lambda_event = {}
+
+    def setUp(self):
+        self.lambda_event = build_lambda_scheduled_event()
+        pass
+
+    def test_scenario_1_compliant_resources(self):
+        describe_images_result = {
+                    'Images': [],
+                    'ResponseMetadata': {'HTTPStatusCode': 200}
+        }
+        ec2_client_mock.describe_images = MagicMock(return_value=describe_images_result)
+        lambda_result = rule.lambda_handler(self.lambda_event, {})
+        expected_response = [
+                	build_expected_response(
+                    		compliance_type='NOT_APPLICABLE',
+                    		compliance_resource_id='N/A',
+                    	    compliance_resource_type=DEFAULT_RESOURCE_TYPE
+                	)
+        ]
+        assert_successful_evaluation(self, lambda_result, expected_response, len(lambda_result))
+
+# Checks for scenario wherein non-compliant resources are present (Two Public Amazon Machine Image)
+class NonCompliantResourcesTest(unittest.TestCase):
+    lambda_event = {}
+
+    def setUp(self):
+        self.lambda_event = build_lambda_scheduled_event()
+        pass
+
+    def test_scenario_2_non_compliant_resources(self):
+            describe_images_result = {
+                'Images': [
+                    {
+                      'ImageId': 'ami-040574eaefd6dc6d4',
+                      'Public': True,
+                      'OwnerId': '123456789012',
+                      'State': 'available'
+                    },
+                    {
+                      'ImageId': 'ami-0a1402bb0642906aa',
+                      'Public': True,
+                      'OwnerId': '123456789012',
+                      'State': 'available'
+                     }
+                ],
+                'ResponseMetadata': {'HTTPStatusCode': 200}
+            }
+            ec2_client_mock.describe_images = MagicMock(return_value=describe_images_result)
+            lambda_result = rule.lambda_handler(self.lambda_event, {})
+            expected_response = [
+                        build_expected_response(
+                                compliance_type='NON_COMPLIANT',
```

Comment:
Same, no need to put all the input variables.

Reply:
removed